EmpireLegends/Player.cpp

```diff
@@ -168,64 +181,201 @@ int Player::MoveOverLand(Territory<Region>* from, Territory<Region>* to, GameMap
 		
-		std::cout << "Successfully destroyed " << number << " armies at " << location->getName() << " ." << std::endl;
-	} else
+		int armyLost = 0;
+		
+
+		//std::cout << "Hewwo" << std::endl;
+		vector<Player*> otherPlayer;
+
+		for (int n = 0; n < currentPlayers.size(); n++) {
+			if (currentPlayers[n]->getName() != getName())
+			{
+				otherPlayer.push_back(currentPlayers[n]);
+			}
+		}
+
+		for (auto ennemi : otherPlayer) {
+			
+			std::cout << ennemi->getName() + " has currently " << ennemi->playerArmies.size() << " armies." << std::endl;
+			for (int i = 0; i < ennemi->playerArmies.size(); i++)
+			{
+				auto armyToDestroy = ennemi->playerArmies[i];
+				if (armyToDestroy->getName() == location->getName())
+				{
+					number -=1;
+					armyLost +=1;
+					ennemi->playerArmies.erase(ennemi->playerArmies.begin() + i);
+					i -= 1;
+				}
+				if (number == 0)
+				{
+					break;
+				}
+			}
+
+			
+			std::cout << ennemi->getName() + " lost " << armyLost << " armies." << std::endl;
+			std::cout << ennemi->getName() + " has now " << ennemi->playerArmies.size() << " armies." << std::endl;
+
+			if (number == 0)
+			{
+				break;
+			}
+		
+		}
+		
+	}
+		
+	else 
 	{
-		std::cout << "Action not permissible (DestroyArmies at " << location->getName() << ")" << std::endl;
+		std::cout << getName() + " couldn't perform this action. (Destroy City)" << std::endl;
 	}
+
 }
 
 // Builds a city at the specified location
 void Player::BuildCity(Territory<Region>* location)
 {
-	// check ownership of region (MUST HAVE AT LEAST ONE ARMY THERE)
+	// check ownership of region (MUST HAVE AT LEAST ONE OF YOUR ARMY THERE)
 	const int unplacedCities = pResources->unplacedCities;
-	if (unplacedCities - 1 >= 0 && (TOTAL_NUM_CITIES - unplacedCities + 1) <= TOTAL_NUM_CITIES && location->getArmyCount() > 0)
+	bool haveOwnArmy = false;
+
+	//auto check = *value;
+	for (auto it : playerArmies) {
+		if (it->getName() == location->getName())
+		{
+			haveOwnArmy = true;
+			break;
+		}
+	}
+	
+
+	if (haveOwnArmy && unplacedCities - 1 >= 0 && (TOTAL_NUM_CITIES - unplacedCities + 1) <= TOTAL_NUM_CITIES && location->getArmyCount() > 0)
 	{
 		pResources->unplacedCities -= 1;
 		// add city at specified location
-		std::cout << "Built a city at " << location->getName() << std::endl;
-	} else if (pResources->unplacedCities - 1 < 0 || (TOTAL_NUM_CITIES - unplacedCities + 1) > TOTAL_NUM_CITIES)
+
+		playerTerritories.push_back( location);
+
+		std::cout << getName() + " built a city at " + location->getName() << std::endl;
+	}
+	else if (pResources->unplacedCities - 1 < 0 || (TOTAL_NUM_CITIES - unplacedCities + 1) > TOTAL_NUM_CITIES)
+	{
+		std::cout << "This action is not permissible due to errors (BuildCity)." << std::endl;
+	}
+	else
 	{
-		std::cout << "This action is not permissible (BuildCity)." << std::endl;
+		std::cout << getName() + " cannot built a city at " + location->getName() + " due to not having an army in this region." << std::endl;
```

Comment:
I feel like these conditional statements can be simplified. We're repeating the condition for placed armies and it seams the two error messages are inverted (?)